### PR TITLE
Add SA10/20/30 series support

### LINKED ISSUE #4: RC5 power on command not working
I am currently testing the project with an Arcam SA10 Integrated Amplifier. While the models are different, the command to turn on the device is the same as the AVR450 which this project was intended for.

as per the [SA-10 Datasheet](https://www.arcam.co.uk/ugc/tor/SA10/Custom%20Installation%20Notes/SH277E_RS232_SA10_SA20_B.pdf):
datasheet

All other RC5 commands when tested work but not the power on. I was able to turn the power on the device by referencing this section of the datasheet:

Screenshot from 2021-08-18 16-52-04

related python code `data` being defined in [`src/arcam/fmj/__init__.py:723`](https://github.com/elupus/arcam_fmj/blob/5b03ef6705748177f4403c8dd3afc10077214a57/src/arcam/fmj/__init__.py#L723):
```py
data = bytes([
    0x21,
    0x01,
    0x00,
    0x01,
    0x01,
    0x0D
])
```

Does anyone know if the RC5 power on command works for their devices?

### PR DESCRIPTION
**Pull request by evanugarte**
Resolves #4

### CODE REVIEW

File: src/arcam/fmj/__init__.py [hunk 1]
```diff
@@ -235,6 +237,7 @@ class SourceCodes(IntOrTypeEnum):
     USB = 0x0F
     STB = 0x10
     GAME = 0x11
+    PHONO = 0x12
```
Comment (evanugarte) [inline src/arcam/fmj/__init__.py:240]: Added new source as `PHONO` is an available for the SA series ([SA series PDF](https://www.arcam.co.uk/ugc/tor/SA10/Custom%20Installation%20Notes/SH277E_RS232_SA10_SA20_B.pdf) page 9)

File: src/arcam/fmj/__init__.py [hunk 2]
```diff
@@ -322,6 +327,8 @@ class DecodeModeMCH(IntOrTypeEnum):
         DecodeMode2CH.MCH_STEREO: bytes([16, 69]),
     },
     (ApiModel.API860_SERIES, 2): {},
+    (ApiModel.APISA_SERIES, 1): {},
+    (ApiModel.APISA_SERIES, 2): {},
```
Comment (evanugarte) [inline src/arcam/fmj/__init__.py:331]: the following are blank as 2CH and MCH decoding are unavailable for the SA models @elupus mentioned in #4

File: src/arcam/fmj/__init__.py [hunk 3]
```diff
@@ -425,6 +460,14 @@ class DecodeModeMCH(IntOrTypeEnum):
     (ApiModel.API860_SERIES, 2): {
         True: bytes([23, 4]),
         False: bytes([23, 5]),
+    },
+    (ApiModel.APISA_SERIES, 1): {
```
Comment (evanugarte) [inline src/arcam/fmj/__init__.py:464]: Power volume and mute RC5 codes are the same for SA models on different zones

File: src/arcam/fmj/__init__.py [hunk 4]
```diff
@@ -387,6 +394,26 @@ class DecodeModeMCH(IntOrTypeEnum):
         SourceCodes.SAT: bytes([16, 20]),
         SourceCodes.VCR: bytes([16, 21]),
         SourceCodes.FOLLOW_ZONE_1: bytes([16, 20])
+    },
+    (ApiModel.APISA_SERIES, 1): {
```
Comment (evanugarte) [inline src/arcam/fmj/__init__.py:398]: The values of these can be verified on the [SA series PDF](https://www.arcam.co.uk/ugc/tor/SA10/Custom%20Installation%20Notes/SH277E_RS232_SA10_SA20_B.pdf) page 19 and [SA30 PDF](http://www.arcam.co.uk/ugc/tor/SA30/Custom%20Installation%20Notes/SH306E_RS232_SA30_4.pdf) page 26

**Review by elupus — APPROVED** [review]
Looks good!

**Comment by elupus** [comment]
@evanugarte any chance you can have a look at: https://github.com/home-assistant/core/issues/84918

**Comment by evanugarte** [comment]
sure thing, i'll try to recreate the issue later today

**Comment by evanugarte** [comment]
to fix, we will need to use a different enumerated type if the device is in the SA series.

in the spec for the Arcam sa10/20 ([pdf](https://www.arcam.co.uk/ugc/tor/SA20/Custom%20Installation%20Notes/SH277E_RS232_SA10_SA20_B.pdf)), it has a different mapping of source to hex value on page 9. the arcam sa30 has the identical mapping on page 10 ([pdf](https://www.arcam.co.uk/ugc/tor/SA30/Custom%20Installation%20Notes/SH306E_RS232_SA30_4.pdf)). in python the mappings become:

```py
class SASeriesSourceCodes(IntOrTypeEnum):
    PHONO = 0x01
    AUX = 0x02
    PVR = 0x03
    AV = 0x04
    STB = 0x05
    CD = 0x06
    BD = 0x07
    SAT = 0x08
    GAME = 0x09
    NET = 0x0B
    USB = 0x0B
    ARC_ERC = 0x0D
    REQUEST_INPUT = 0xF0
```

what would be the best way to do this? the `Client` class would need to know what the type of Arcam device it is interacting with. perhaps passing in the device series to the constructor (defined in [src/arcam/fmj/__init__.py:86](https://github.com/elupus/arcam_fmj/blob/37015e1297f6e01acaa126923eda42d3eb7eceeb/src/arcam/fmj/__init__.py#L86))? then, if the series is SA, we can use this modified enum of source mappings.

**Comment by elupus** [comment]
I was pondering removing the value in the enum and maintaining a separate dict that maps byte values to enum values.

**Comment by evanugarte** [comment]
maybe we could pass in the `ApiModel` enum as an optional parameter, and if the model is `APISA_SERIES` we use the different mapping

**Comment by elupus** [comment]
Model is already known by the State class. It's checked on startup. So it's not that hard to remove the byte numbers from the Source and keep that as a separate map that is selected based on series.

**Comment by evanugarte** [comment]
Intend to work on this over the weekend

**Comment by evanugarte** [comment]
i plan to open a pr similar to #7, as the SA series supports non-rc5 commands to change source later today

**Pull request merged**